### PR TITLE
Breaking changes: the getResponse() function from \WindowsAzure\Common\Internal\Http\IHttpClient is replaced by sendAndGetResponse()

### PR DESCRIPTION
**Pull request by sergey-shandar**
We plan to remove PEAR libraries such as `HTTP_Request2` (Issue #797). So we need to replace functions which return types from the PEAR libraries, such as `HTTP_Request2_Response`. This PR is one of the first steps to do so.

### CODE REVIEW

File: src/Common/Internal/Http/HttpClient.php [hunk 1]
```diff
@@ -240,18 +234,18 @@ public function setPostParameters($postParameters)
...
      * @param IUrl  $url     Request url.
      *
      * @throws WindowsAzure\Common\ServiceException
-     *
-     * @return string The response body
+     * 
+     * @return \HTTP_Request2_Response The response.
      */
-    public function send($filters, $url = null)
+    public function sendAndGetResponse($filters, $url = null)
```
Comment (yaqiyang) [inline src/Common/Internal/Http/HttpClient.php]: I don't see any good reason for renaming this function

Comment (yaqiyang) [inline src/Common/Internal/Http/HttpClient.php]: that is fine

File: src/Common/Internal/Http/HttpClient.php [hunk 2]
```diff
@@ -279,23 +273,38 @@ public function send($filters, $url = null)
...
+     *                       send and then applied to the response.
+     * @param IUrl  $url     Request url.
+     *
+     * @throws WindowsAzure\Common\ServiceException
+     *
+     * @return string The response body
+     */
+    public function send($filters, $url = null)
+    {
+        return $this->sendAndGetResponse($filters, $url)->getBody();
```
Comment (yaqiyang) [inline src/Common/Internal/Http/HttpClient.php]: will the response from sendAndGetResponse() ever be null

File: tests/unit/WindowsAzure/Common/Internal/Http/HttpClientTest.php [hunk 3]
```diff
@@ -228,6 +228,23 @@ public function testSetHeaderExistingHeaderAppend()
...
+        // Setup
+        $channel = new HttpClient();
+        $url = new Url('http://example.com/');
+        $channel->setExpectedStatusCode('200');
+        
+        // Test
+        $response = $channel->sendAndGetResponse(array(), $url);
+
+        // Assert
+        $this->assertInstanceOf('\HTTP_Request2_Response', $response);
```
Comment (yaqiyang) [inline tests/unit/WindowsAzure/Common/Internal/Http/HttpClientTest.php]: this does not seem right. The instance is not of HTTP_Request2 anymore

File: src/Common/Internal/Http/HttpClient.php (hunk 2 above)
Comment (sergey-shandar) [inline src/Common/Internal/Http/HttpClient.php]: It shouldn't. The sendAndGetResponse() should throw an exception if there is any problem.

**Pull request merged**